In [1]:
import pandas as pd

import plotly_express as px

In [2]:
df = pd.read_pickle('../data/in_progress/pickled_data.pkl')

df

reference       date navArea  subreg       hostility  \
0     2022-114 2022-04-28      XI      91  No information   
1     2022-113 2022-04-27      XI      71        Boarding   
2     2022-112 2022-04-26       V      24        Boarding   
3     2022-111 2022-04-26    VIII      63  No information   
4     2022-110 2022-04-25       I      44         Protest   
...        ...        ...     ...     ...             ...   
8614    1980-1 1980-01-11     III      56         PIRATES   
8615    1979-2 1979-10-10      XI      93  No information   
8616    1979-1 1979-02-24      XI      93  No information   
8617    1978-2 1978-05-16      XI      92         PIRATES   
8618    1978-1 1978-05-01      XI      92         UNKNOWN   

                                       victim  \
0     Marshall Islands-flagged tanker SEXTANS   
1                                Bulk carrier   
2                                Bulk carrier   
3       Singapore-flagged tanker STI MAGISTER   
4           Hong Kong-flagged tanker UST LUGA   
...                                       ...   
8614               INSTERGENERAL R. GUMUSPALA   
8615                           No information   
8616                           No information   
8617                          MERCHANT VESSEL   
8618                           UNKNOWN VESSEL   

                                            description   latitude   longitude  
0     PHILIPPINES: On 28 April, at 0345 local time, ...  13.716667  121.033333  
1     INDONESIA On 27 April, at 0226 local time, a r...   1.266667  104.283333  
2     BRAZIL: On 26 April, at 0400 local time, three...  -0.066667  -51.100000  
3     BANGLADESH: On 26 April, at 0001 local time, a...  22.216667   91.716667  
4     NORWAY: On the morning of 25 April, a group of...  59.316667   10.500000  
...                                                 ...        ...         ...  
8614  Instergeneral R. Gumuspala, a Turkish-flag gen...  34.000000   34.000000  
8615  SOUTH CHINA SEA, PARACEL ISLANDS.             ...  17.000000  112.000000  
8616  GULF OF TONKIN.                               ...  19.000000  108.000000  
8617  SULU SEA: Lloyd's List of 15 May reports a 5,0...   8.000000  120.000000  
8618  SOUTH SULU SEA                                ...   6.000000  119.500000  

[8613 rows x 9 columns]

In [3]:
df['hostility'].nunique()

431

In [4]:
df['victim'].nunique()

1587

#### 431 types of hostilities, and 1587 types of victims both seem a bit ... excessive.
And it's not like it can be argued these are the total number of cases because there are 8619 events. This isn't critical for now though so let's move on.

### Checking for additional suprizes

In [5]:
df['navArea'].unique()

array(['XI', 'V', 'VIII', 'I', 'III', 'II', 'IV', 'XVI', 'VII', 'IX', 'X',
       'XII', 'No information', 'XIV', 'XIII', 'VI', 'XV'], dtype=object)

In [6]:
df.loc[df['navArea'] == 'No information']

reference       date         navArea  subreg   hostility  \
1046  2017-321 2017-11-30  No information      62     Robbers   
1877   2015-85 2015-04-24  No information      63     Robbers   
8063    1997-4 1997-01-22  No information      22  ASSAILANTS   

               victim                                        description  \
1046   Sailing Vessel  CAPE VERDE: On 30 November, a Netherlands-flag...   
1877   Product Tanker  BANGLADESH: On 24 April, a duty crewman aboard...   
8063  MERCHANT VESSEL  South America                                 ...   

       latitude  longitude  
1046  14.900000  23.500000  
1877  27.116667  91.783333  
8063  -3.462500 -72.607500

The entries have lat/long values so I'm not worried about it and can correct this in the future. I'm curious about why it wasn't entered in the first place though. 
<br><br>
Subreg has no issues as you can see below.

In [7]:
df['subreg'].unique()

array([91, 71, 24, 63, 44, 92, 55, 57, 37, 13, 28, 22, 25, 26, 72, 62, 61,
       73, 51, 82, 93, 18, 94, 56, 52, 21, 35, 27, 53, 95, 83, 74, 81, 54,
       96, 97, 36, 12, 23, 11])

In [13]:
fig = px.scatter_mapbox( data_frame=df, lat="latitude", lon="longitude",
        mapbox_style="carto-darkmatter", color_discrete_sequence=["gold"], 
        zoom=1, height=700, width=1500, center= {'lat': 16 ,'lon': 99})

fig.show()